modules/gtk/gtk_mod.c

```diff
@@ -771,30 +771,22 @@ static gboolean status_icon_on_button_press(GtkStatusIcon *status_icon,
 
 int gtk_mod_connect(struct gtk_mod *mod, const char *uri)
 {
-	struct mbuf *uribuf = NULL;
-	char *uri_copy = NULL;
+	char *uric = NULL;
+	struct pl pl;
```

Comment:
I think we should focus on menu.so and ignore gtk.so in this patch.

Reply:
The second API change is `account_uri_complete()` which simplifies the code like this:
```patch
-       uribuf = mbuf_alloc(64);
-       if (!uribuf)
-               return ENOMEM;
-
-       err = account_uri_complete(ua_account(mod->ua_cur), uribuf, uri);
-       if (err)
-               goto out;
-
-       uribuf->pos = 0;
-       err = mbuf_strdup(uribuf, &uri_copy, uribuf->end);
+       pl_set_str(&pl, uri);
+       err = account_uri_complete(ua_account(mod->ua_cur), &uric, &pl);
```